### DISCUSSION TITLE
How do I use DMA features on the synopsys platform? Does tinyusb support DMA?

### DISCUSSION THREAD

**Discussion by jeffery88** [hathach/tinyusb#2190] [Q&A]
Our platform is baseed on usb ip synopsys\dwc2. The current usbd and cdc seem to be all based on fifo operations. How do I use DMA features on the synopsys platform? Does tinyusb support DMA?

**Comment by cleverca22** [comment]
ive been implementing dwc2 host mode on the rpi, and have also run into issues when i flipped dma on: https://github.com/hathach/tinyusb/discussions/2189

the 2 main problems ive had are:
* needing all buffers to be 32bit aligned, and nothing in the last 1-3 bytes, so a 32bit write doesnt trash something
* the interrupts are firing slightly differently, and that broke the entire stack

https://github.com/librerpi/lk-overlay/tree/master/platform/bcm28xx/dwc2 is where all of my source has been going
